src/twisted/names/dns.py

```diff
@@ -2005,13 +2093,88 @@ class Record_SPF(Record_TXT):
     @ivar data: Freeform text which makes up this record.
 
     @type ttl: L{int}
-    @ivar ttl: The maximum number of seconds which this record should be cached.
+    @ivar ttl: The maximum number of seconds
+               which this record should be cached.
     """
     TYPE = SPF
     fancybasename = 'SPF'
 
 
 
+@implementer(IEncodable, IRecord)
+class Record_TSIG(tputil.FancyEqMixin, tputil.FancyStrMixin):
+    """
+    A transaction signature, encapsulated in a RR, as described
+    in U{RFC 2845 <https://tools.ietf.org/html/rfc2845>}.
+
+    @type algorithm: L{Name}
```

Comment:
what type is Name?

also, to match the RFC, shouldn't this be algorithmName? as otherwise, instead of 'timeSigned' we can have just 'signed'

Reply:
`Name` is a class in the same module. The apidocs seem to find and link to it correctly as is. (It corresponds to a domain name.)

I would normally be in favor of shortening `timeSigned` but I think that changing `timeSigned` to `signed` would be confusing — there are too many things that could indicate (time signed, whether it's signed, the item being signed, etc). Calling it `time` would be OK, perhaps?

Reply:
Yes. Sorry for the noise with `Name`.
As for naming the variables, I am fine with any other name if you think that it would make it easier to use.
As a non DNS guy, my initial thought was that keeping the names as close as possible to the RFC would reduce the confusion of whether something is associated with the RFC or is extra API.

Reply:
"time" is the name of an existing module, so not a good choice. timeSigned is fine. It's the time it was signed, so timeSigned "reads" well.